clang/lib/CIR/Lowering/ThroughMLIR/LowerCIRToMLIR.cpp

```diff
@@ -320,6 +320,61 @@ class CIRExpOpLowering : public mlir::OpConversionPattern<mlir::cir::ExpOp> {
   }
 };
 
+static mlir::Value createIntCast(mlir::ConversionPatternRewriter &rewriter,
+                                 mlir::Value src, mlir::Type dstTy,
+                                 bool isSigned = false) {
+  auto srcTy = src.getType();
+  assert(isa<mlir::IntegerType>(srcTy));
+  assert(isa<mlir::IntegerType>(dstTy));
+
+  auto srcWidth = srcTy.cast<mlir::IntegerType>().getWidth();
+  auto dstWidth = dstTy.cast<mlir::IntegerType>().getWidth();
+  auto loc = src.getLoc();
+
+  if (dstWidth > srcWidth && isSigned)
+    return rewriter.create<mlir::arith::ExtSIOp>(loc, dstTy, src);
+  else if (dstWidth > srcWidth)
+    return rewriter.create<mlir::arith::ExtUIOp>(loc, dstTy, src);
+  else if (dstWidth < srcWidth)
+    return rewriter.create<mlir::arith::TruncIOp>(loc, dstTy, src);
+  else
+    return rewriter.create<mlir::arith::BitcastOp>(loc, dstTy, src);
```

Comment:
No tests cover this branch. Consider adding a test for it.

Besides, CIR currently does not generate MLIR integer types with signedness semantics. Thus the bitcast here should be unnecessary. But this involves modifying existing code so it's OK if you don't address this in this PR.

Reply:
You are right, I have added test coverage for the last branch